### PR TITLE
Material 3 DropdownMenu Skiko support

### PR DESCRIPTION
**Pull request by eygraber**
## Proposed Changes

  - Replace `Popup` from desktop and uikit with a `Popup` in skiko
  - Add a `DropdownMenu` to material3 for skiko

## Testing

Test: Copied the `DesktopMenuTest` from material into material3; kept `DesktopPopupTest` in `ui/desktopTest` which now uses `Popup` from skiko, since I couldn't get the tests to pass in `ui/skikoTest` (got everything to compile, but couldn't figure out how to idle with `ImageComposeScene`; maybe related to https://github.com/JetBrains/compose-jb/issues/1866)

## Issues Fixed

Addresses https://github.com/JetBrains/compose-jb/issues/2037

Possibly supersedes https://github.com/JetBrains/androidx/pull/328

CLA is signed

### CODE REVIEW

File: compose/material3/material3/src/skikoMain/kotlin/androidx/compose/material3/SkikoMenu.skiko.kt [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+    expandedStates.targetState = expanded
+
+    if (expandedStates.currentState || expandedStates.targetState) {
+        val transformOriginState = remember { mutableStateOf(TransformOrigin.Center) }
+        val density = LocalDensity.current
+        // The original [DropdownMenuPositionProvider] is not yet suitable for large screen devices,
+        // so we need to make additional checks and adjust the position of the [DropdownMenu] to
+        // avoid content being cut off if the [DropdownMenu] contains too many items.
+        // See: https://github.com/JetBrains/compose-jb/issues/1388
+        val popupPositionProvider = SkikoDropdownMenuPositionProvider(
```
Comment (eygraber) [inline compose/material3/material3/src/skikoMain/kotlin/androidx/compose/material3/SkikoMenu.skiko.kt:111]: This has diverged from `DropdownMenuPositionProvider` which now handles LTR. If you want, I can make a followup PR that ports that behavior here and into the material `DesktopDropdownMenuPositionProvider`.

**Comment by eygraber** [comment]
@igordmn any chance this makes it into 1.3.0?

**Comment by igordmn** [comment]
Hi, thanks for the changes! I will look at your PR's soon.

In the meantime, could you look at [this comment](https://github.com/JetBrains/androidx/pull/326#issuecomment-1312220826), and do what is asked there (sign CLA, put "CLA is signed" in the PR description), and agree on the terms?

**Comment by eygraber** [comment]
Updated description regarding CLA and I agree that author fields can be lost

**Comment by eygraber** [comment]
Do I need to do this for all my open PRs?

**Comment by igordmn** [comment]
> Do I need to do this for all my open PRs?

Yes, just add "CLA is signed" into them

File: compose/material3/material3/src/skikoMain/kotlin/androidx/compose/material3/SkikoMenu.skiko.kt (hunk 1 above)
Comment (igordmn) [reply compose/material3/material3/src/skikoMain/kotlin/androidx/compose/material3/SkikoMenu.skiko.kt:111]: That would be great, thanks.

P.S. Doesn't it handle LTR already? We have `if (layoutDirection == LayoutDirection.Ltr)` in the DesktopDropdownMenuPositionProvider/SkikoDropdownMenuPositionProvider

Comment (eygraber) [reply compose/material3/material3/src/skikoMain/kotlin/androidx/compose/material3/SkikoMenu.skiko.kt:111]: Oh interesting, guess I missed that :sweat_smile:

**Pull request merged**